src/components/atoms/ControlledFields/CheckboxWithLabel/CheckboxWithLabel.tsx

```diff
@@ -0,0 +1,44 @@
+import { Checkbox, FormControlLabel } from '@mui/material';
+import { useController } from 'react-hook-form';
+
+import { CheckedIcon } from '../../Icons/CheckedIcon';
+import { CheckboxSpan } from './styles';
+import { IProps } from './types';
+
+export const CheckboxWithLabel = <T extends object>({
+  name,
+  control,
+  label,
+  onChange,
+  ...props
+}: IProps<T>) => {
+  const { field } = useController({
+    name,
+    control,
+  });
+
+  return (
+    <FormControlLabel
+      label={label}
+      control={
+        <Checkbox
+          {...field}
+          checked={field.value}
+          onChange={({ target: { checked: targetChecked } }) => {
+            field.onChange(targetChecked);
+            if (onChange) {
+              onChange(targetChecked);
```

Comment:
Why do you need a duplicate `onChange` method?
In my point of view, it is no need for a template.😉

Reply:
it's used for extending this component for cases when this input depends on some logic